### ISSUE TITLE
fix: AAC files using M4A wrapper not loading properly

### ISSUE THREAD

**Issue by mortgagemetrix** [ubuntu-flutter-community/musicpod#1087]
AAC files are not scanned properly. Most encoders save AAC audio in the M4A wrapper but Musticpod does not recognize these files. If I manually change the file extension to MP4, the file is recognized and able to be played. MP4 wrapper extension is typically used for video, not audio files.

**Comment by Feichtmeier** [comment]
> AAC files are not scanned properly. Most encoders save AAC audio in the M4A wrapper but Musticpod does not recognize these files. If I manually change the file extension to MP4, the file is recognized and able to be played. MP4 wrapper extension is typically used for video, not audio files.

hmmmm, well
the library I use for reading metadata supports flac, mp3, ogg, opus and mp4
Sure I could add more allowed file extensions but then the music files are just loaded with there file types
generally the local audio feature of musicpod (remember radio and podcasts have the same prio) is designed around users with tagged libraries. I don't want to tap into this gray area of creating some database of "might be the right info" for each file, just because people dont buy music anymore. (mp3 downloads have everything tagged perfectly)

anyways, do you know if aac files have metadata?

**Comment by mortgagemetrix** [comment]
No, M4A (a file format for AAC audio encapsulated in MP4) does not use ID3v2 tags for metadata. Instead, it uses MP4 metadata atoms, which are part of the ISO Base Media File Format (commonly used for MP4 files).

FYI, Quod Libet doesn't really get this right either. It correctly maps `©ART` to `TPE1` but treats `aART` as distinct from `TPE2`, mapping `aART` to `Album Artist` and `TPE2` to `Performer`. iTunes and Picard both treat these tags as interchangeable.

**Key Differences**
ID3v2 Tags:
- Used in formats like MP3.
- Organized into "frames" (e.g., TPE1, TALB, etc.).
- Typically placed at the beginning of the file.

MP4 Metadata Atoms:
- Used in M4A, MP4, and similar formats.
- Organized into "atoms" (e.g., ©ART, ©alb, aART).
- Embedded within the file structure.

| MP4 Atom  | Description                | ID3v2 Equivalent |
|-----------|----------------------------|----------------- |
| `©ART`    | Artist                     | `TPE1` |
| `©alb`    | Album                      | `TALB` |
| `aART`    | Album Artist               | `TPE2` |
| `©nam`    | Track Title                | `TIT2` |
| `trkn`    | Track Number               | `TRCK` |
| `disk`    | Disc Number                | `TPOS` |
| `©gen`    | Genre                      | `TCON` |
| `©day`    | Year                       | `TYER` |
| `soar`    | Sort Order for Artist      | `TSOP` |
| `soaa`    | Sort Order for Album Artist| `TSO2` |
| `soal`    | Sort Order for Album       | `TSOA` |
| `sonm`    | Sort Order for Track Title | `TSOT` |

**Comment by Feichtmeier** [comment]
@mortgagemetrix that those files are not played shoud now be fixed since #1167 is fixed
so what is missing is reading the metadata from aac or m4a files now, right?

**Comment by mortgagemetrix** [comment]
Musicpod could already open AAC files, but you have to change the extension from .m4a to .mp4. An easy short term fix would be to treat .m4a as the same as mp4. The tags seem to work mostly correctly as well. It's showing both album artist and artist as Nirvana in my case. Apple Music correctly shows the album artist as Various Artists and categorizes the album as Compilation (note the flag, likely apple specific). I'm away from my Ubuntu machine so I tested 2.9.0 on Mac OS.

I've attached the output of the `mediainfo` command line app to show what the file says. Note the difference from Apple Music is because the file was run through MusicBrainz Picard and it change the name from "Verse Chorus Verse" to "Sappy". Still shows "Various Artists" which is the relevant bit. I don't know if Picard set the compilation flag or if Apple Music did.
[sappy.md](https://github.com/user-attachments/files/19558178/sappy.md)

<img width="517" alt="Image" src="https://github.com/user-attachments/assets/01b43903-9a63-418c-ac82-2fbf536fdcdc" />
<img width="730" alt="Image" src="https://github.com/user-attachments/assets/adfe36dd-42f2-4a7e-9310-2d52f5876002" />
<img width="604" alt="Image" src="https://github.com/user-attachments/assets/f32448a7-6ad9-49fd-a96f-e3db4aca5a68" />
<img width="853" alt="Image" src="https://github.com/user-attachments/assets/1e86f33b-6ac0-47e0-904d-e457bb51d079" />
